op-service/sources/l1_beacon_client.go

```diff
@@ -267,17 +296,37 @@ func (cl *L1BeaconClient) GetBlobSidecars(ctx context.Context, ref eth.L1BlockRe
 // blob's validity by checking its proof against the commitment, and confirming the commitment
 // hashes to the expected value. Returns error if any blob is found invalid.
 func (cl *L1BeaconClient) GetBlobs(ctx context.Context, ref eth.L1BlockRef, hashes []eth.IndexedBlobHash) ([]*eth.Blob, error) {
-	blobSidecars, err := cl.GetBlobSidecars(ctx, ref, hashes)
+	if len(hashes) == 0 {
+		return []*eth.Blob{}, nil
+	}
+	slot, err := cl.timeToSlot(ctx, ref.Time)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get blob sidecars for L1BlockRef %s: %w", ref, err)
+		return nil, err
 	}
-	blobs, err := blobsFromSidecars(blobSidecars, hashes)
+	blobs, err := cl.cl.BeaconBlobs(ctx, slot, hashes)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get blobs from sidecars for L1BlockRef %s: %w", ref, err)
+		// We would normally check for an explicit error like "method not found", but the Beacon
+		// API doesn't standardize such a response. Thus, we interpret all errors as
+		// "method not found" and fall back to fetching sidecars.
+		blobSidecars, err := cl.getBlobSidecars(ctx, slot, hashes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get blob sidecars for L1BlockRef %s: %w", ref, err)
+		}
+		blobs, err := blobsFromSidecars(blobSidecars, hashes)
```

Comment:
I suggest we set a state variable like `cl.useSidecars` in the client if it ever had to fall back to querying sidecars and that was successful, so the client doesn't every time query the `blobs` endpoint again that's already proven not to work. If encountering an error, it can set this back to `false` so that an endpoint that suddenly upgrades to have the new endpoint would also keep working.
```suggestion
		blobSidecars, err := cl.getBlobSidecars(ctx, slot, hashes)
		if err != nil {
			cl.useSidecars = false
			return nil, fmt.Errorf("failed to get blob sidecars for L1BlockRef %s: %w", ref, err)
		}
		cl.useSidecars = true
		blobs, err := blobsFromSidecars(blobSidecars, hashes)
```
This state variable would be checked above before calling `cl.cl.BeaconBlobs(ctx, slot, hashes)`